src/changeset.js

```diff
@@ -153,7 +153,7 @@ module.exports = (logger, s3, options) => Promise.all([
         if (!existence.remotely) {
           added.push(info(name, false, options));
         } else if (!existence.locally) {
-          if (options.delete || options.softDelete) {
+          if ((options.delete || options.softDelete) && !micromatch.isMatch(name, options.retain)) {
```

Comment:
Please add a log message (I guess debug level is sufficient) when deleted objects will be retained.

Reply:
Done.